### PR TITLE
RCA Variables -> add new variables

### PR DESCRIPTION
**Pull request by JoeArmani**
Related: https://github.com/episphere/connect/issues/866

•Add new variables and validation to existing RCA Variables implementation.

### CODE REVIEW

File: utils/shared.js [hunk 1]
```diff
@@ -1260,6 +1265,22 @@ const validateCancerOccurrence = (cancerSitesObject) => {
     return { error: hasError, message: hasError ? otherCancerSiteErrorMessage : '', data: [] };
 }
 
+/**
+ * Rules: if vitalStatusCategorical is 'alive' at chart review (114227122: 337516613), participant must be aware of diagnosis (844209241: 353358909). Else, block API request.
+ * If vitalStatusCategorical is 'dead' or 'unknown' (114227122: 646675764 or 178420302), participant awareness can be yes, no, or unknown (844209241: 353358909 or 104430631 or 178420302).
+ * @param {number} vitalStatusCategorical - the participant's vital status (conceptID).
+ * @param {number} participantDiagnosisAwareness - the participant's awareness of diagnosis (conceptID).
```
Comment (jhflorey) [inline utils/shared.js:1272]: NIT: Should we define `return`

**Review by jhflorey — APPROVED** [review]

File: updateParticipantData.json [hunk 2]
```diff
@@ -535,6 +535,23 @@
 		"dataType": "object",
 		"mustExist": false
 	},
+	"637153953[].844209241": {
```
Comment (we-ai) [inline updateParticipantData.json:538]: Why using `[]` after `637153953`?

Comment (JoeArmani) [reply updateParticipantData.json:538]: It's for validation of data in a variable length array. The `[]` part enforces the array data structure for validation rules so we can validate any index. Then, during validation, we remove the actual index for an equal comparison.
 
Rules format (JSON): https://github.com/episphere/connectFaas/blob/fa33180d31b0d1249502b295787ca26c80b380b6/updateParticipantData.json#L520
 
Validation (stripping the index to enforce the array structure): https://github.com/episphere/connectFaas/blob/fa33180d31b0d1249502b295787ca26c80b380b6/utils/sites.js#L399

Comment (we-ai) [reply updateParticipantData.json:538]: I see it now.

File: utils/shared.js (hunk 1 above)
Comment (JoeArmani) [reply utils/shared.js:1272]: Good gatch, thanks. Updated in this function and `validateCancerOccurrence`

**Review by we-ai — APPROVED** [review]

**Pull request merged**